### PR TITLE
allow scala tests to omit verbose test logs and xml files

### PR DESCRIPTION
**Pull request by natansil**
changed `scala_junit_test` runner to `bazel_test_runner` instead of `JUnitCore`. this allows for complete logs and xml output to be generated. the `bazel_test_runne`r source code was copied to `rules_scala` in order to depend on it in the `scala_junit_test` rule.
This is a temporary measure, where the duplicate code can be deleted once java_test will be able to depend on scala_library.

This PR will allow to resolve issue https://github.com/bazelbuild/rules_scala/issues/211 (test.xml output from scala_junit_test is lacking in detail)

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by johnynek** [comment]
Thanks for the PR.

I'm concerned about copying 8k lines from bazel. Is there any way we can:

1. use bazel code as a remote repository?
2. could we minimize this somewhat? The scala_test xml output is only ~500 lines: https://github.com/bazelbuild/rules_scala/blob/master/scala/support/JUnitXmlReporter.scala

**Comment by ittaiz** [comment]
@johnynek I'm fairly certain we can't use it since the targets are not public. I asked Ulf about it sometime ago and he said to copy the code (he might thought it's less code though).
Main reason I think this is ok is because AFAIU @sdtwigg will finish his work very soon and enable us to delete all of this code since we'll be using java_test which has access to this Runner.

File: WORKSPACE [hunk 1]
```diff
@@ -29,3 +29,17 @@ maven_jar(
   sha1 = "e5b3e2753d0817b622c32aedcb888bcf39e275b4")
 
 
+# dependencies needed for BazelTestRunner
+maven_server(
+    name = "one_hippo",
```
Comment (ittaiz) [inline WORKSPACE]: why do we need this server? doesn't maven central have jsr305?

File: WORKSPACE [hunk 2]
```diff
@@ -29,3 +29,17 @@ maven_jar(
...
+    url = "http://maven.onehippo.com/maven2/",
+)
+
+maven_jar(
```
Comment (ittaiz) [inline WORKSPACE]: @johnynek This should probably move to the *_repositories pattern and bind, right?

**Comment by ittaiz** [comment]
test this please (cibot)

**Comment by natansil** [comment]
@johnynek, I've looked into minimizing the code. Unfortunately not a lot can be removed...
An added bonus is getting the tests sharding feature of the bazel test runner... 
I think it's the best temporary solution until the switch to java_test

**Comment by ittaiz** [comment]
test this please (ci)

**Comment by chaoren** [comment]
@ittaiz 
> I'm fairly certain we can't use it since the targets are not public.

Can we just download the files and make our own targets? Like we did with https://github.com/bazelbuild/rules_scala/blob/a94b6d0690f48cd41d3649d8aab9a7fd1491b76e/scala/scala.bzl#L753.

**Comment by ittaiz** [comment]
@chaoren we're not generating any targets in the stub example. We're just using a source file. In this case we need a binary. I've tried looking in the zip and I don't see the runner jar in it. Am I looking in the wrong place? If there is an easy archive we can depend on which exposes the jar that's definitely an option

**Comment by chaoren** [comment]
@ittaiz 
> I've tried looking in the zip and I don't see the runner jar in it. 

But the sources to build the jar is available from https://github.com/bazelbuild/bazel/archive/0.5.1.tar.gz, no?

Instead of just copying the sources, why not refer to them directly from the package? Just create an new_http_archive, and our own BUILD file for it.

**Comment by natansil** [comment]
@chaoren, from what I see in `new_http_archive` documentation, only the source files are exposed. Do you suggest to merge all of the _bazel_ runner related BUILD files into one _rules_scala_ BUILD file?

**Comment by chaoren** [comment]
@natansil yes, create our own BUILD file for those sources.

**Comment by natansil** [comment]
@chaoren , it will take me a few days to get to this. so we have 3 options:
1. someone else try to create own BUILD file
2. merge the current solution
3. wait for when I have more availability for this

**Comment by chaoren** [comment]
@natansil 
> 2. merge the current solution

Please don't.

I'll do it.

**Comment by chaoren** [comment]
Okay, so after playing around a bit, I couldn't find a way across the package boundary created by the existing BUILD files. (Tried some weird tricks with `strip_prefix`.)

But, I did find the the jar we need at `@bazel_tools//tools/jdk:TestRunner_deploy.jar`. However, it is not publicly visible, unlike the [other deploy jars next to it](https://github.com/bazelbuild/bazel/blob/master/tools/jdk/BUILD). I'll send a pull request upstream to expose test runner for our use. If they accept, then problem solved.

If they don't accept, then we can still just build the jar from source and check it in. Of course there's the security implication of checking in a jar, but since bazel builds should be reproducible, the reviewer (or whoever cares) can just verify by building the jar themselves and comparing.

I think either solution is superior and more maintainable than this PR.

**Comment by ittaiz** [comment]
Thanks!
Let's wait and see the response on your PR.
I agree both are probably better.

**Comment by natansil** [comment]
superseded by https://github.com/bazelbuild/rules_scala/pull/228

**Pull request closed without merging**